hazelcast/src/test/java/com/hazelcast/replicatedmap/VectorClockTest.java

```diff
@@ -0,0 +1,267 @@
+    public void testEmptyClockShouldNotHappensBeforeItself() throws Exception {
+        VectorClockTimestamp clock = new VectorClockTimestamp();
+        assertFalse(VectorClockTimestamp.happenedBefore(clock, clock));
+    }
+
+    @Test
+    public void testIdenticalClocksShouldNotHaveHappensBeforeRelation() throws Exception {
+        VectorClockTimestamp clock1 = new VectorClockTimestamp();
+        VectorClockTimestamp clock2 = new VectorClockTimestamp();
+        MemberImpl localMember = new MemberImpl(new Address("127.0.0.1", 0));
+        clock1 = clock1.incrementClock(localMember);
+        clock2 = clock2.incrementClock(localMember);
+        assertFalse(VectorClockTimestamp.happenedBefore(clock1, clock2));
+        assertFalse(VectorClockTimestamp.happenedBefore(clock2, clock1));
+    }
+
+    @Test
+    public void testClockShouldHappensBeforeWithIdenticalClockWithOneMoreUpdate() throws Exception {
+        VectorClockTimestamp clock1 = new VectorClockTimestamp();
+        VectorClockTimestamp clock2 = new VectorClockTimestamp();
+        MemberImpl localMember = new MemberImpl(new Address("127.0.0.1", 0));
+        clock1 = clock1.incrementClock(localMember);
+        clock2 = clock2.incrementClock(localMember);
+        clock2 = clock2.incrementClock(localMember);
+        assertTrue(VectorClockTimestamp.happenedBefore(clock1, clock2));
+    }
+
+    @Test
+    public void testHappensBefore() throws Exception {
+        VectorClockTimestamp clock1 = new VectorClockTimestamp();
+        VectorClockTimestamp clock2 = new VectorClockTimestamp();
+        MemberImpl member1 = new MemberImpl(new Address("127.0.0.1", 0));
+        MemberImpl member2 = new MemberImpl(new Address("127.0.0.2", 0));
+        MemberImpl member3 = new MemberImpl(new Address("127.0.0.3", 0));
+        clock1 = clock1.incrementClock(member2);
+        clock1 = clock1.incrementClock(member2);
+
+        clock2 = clock2.incrementClock(member1);
+        clock2 = clock2.incrementClock(member2);
+        clock2 = clock2.incrementClock(member2);
+        clock2 = clock2.incrementClock(member3);
+
+        // we need to represent happens before with an enum not with boolean
+        assertTrue(VectorClockTimestamp.happenedBefore(clock1, clock2));
+        assertFalse(VectorClockTimestamp.happenedBefore(clock2, clock1));
+
+
+    }
+
+    @Test
+    public void testConcurrentEventsShouldNotHaveHappensBeforeRelation() throws Exception {
+        VectorClockTimestamp clock1 = new VectorClockTimestamp();
+        VectorClockTimestamp clock2 = new VectorClockTimestamp();
+        MemberImpl member1 = new MemberImpl(new Address("127.0.0.1", 0));
+        MemberImpl member2 = new MemberImpl(new Address("127.0.0.2", 0));
+        clock1 = clock1.incrementClock(member1);
+        clock2 = clock2.incrementClock(member2);
+        // we need to represent happens before with an enum not with boolean
+        assertFalse(VectorClockTimestamp.happenedBefore(clock1, clock2));
```

Comment:
I would introduce a custom assert assertHappensBefore and assertNotHappensBefore.